packages/x-date-pickers/src/internals/hooks/useDefaultReduceAnimations.ts

```diff
@@ -2,8 +2,15 @@ import useMediaQuery from '@mui/material/useMediaQuery';
 
 const PREFERS_REDUCED_MOTION = '@media (prefers-reduced-motion: reduce)';
 
+// detect if user agent has Android version < 10 or iOS version < 13
+const mobileVersionMatches =
+  typeof navigator !== 'undefined' && navigator.userAgent.match(/android\s(\d+)|OS\s(\d+)/i);
+const androidVersion =
+  mobileVersionMatches && mobileVersionMatches[1] ? parseInt(mobileVersionMatches[1], 10) : null;
+const iOSVersion =
+  mobileVersionMatches && mobileVersionMatches[2] ? parseInt(mobileVersionMatches[2], 10) : null;
 export const defaultReduceAnimations =
-  typeof navigator !== 'undefined' && /(android)/i.test(navigator.userAgent);
+  (androidVersion && androidVersion < 10) || (iOSVersion && iOSVersion < 13) || false;
 
 export const useDefaultReduceAnimations = () => {
```

Comment:
I wonder if 
```suggestion
export const useDisableAnimations = () => {
```

wouldn't be clearer, as far as I understand it, we are not reducing animations but disabling them.